src/main/java/org/commonjava/util/gateway/metrics/response/ResponseCountAspect.java

```diff
@@ -0,0 +1,34 @@
+package org.commonjava.util.gateway.metrics.response;
+
+import org.commonjava.util.gateway.metrics.response.component.ResponseCount;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+@ResponseCount
+@Interceptor
+public class ResponseCountAspect {
```

Comment:
I am not sure this interceptor can work under the Mutiny structure. The Mutiny REST endpoint returns a Uni which is handled by back-end threads. In other words, the REST method will return very quickly but the real action is executed asynchronously. You may do a test to verify. For example, use a GET method which returns a Uni and inside the function it sleeps 10s to see if the interceptor can print the right latency.

Reply:
Thanks for your advice. Let me have a try.

Reply:
I have a question: the real action is executed in the [enqueue() of WebClientAdapter.java](https://github.com/Commonjava/gateway/blob/6f0296c1d106c80ae6f01bf266694ea9f68bb924/src/main/java/org/commonjava/util/gateway/util/WebClientAdapter.java#L305)?

Reply:
I'd suggest use some simple code to test it. Maybe get some sample code from Quarkus docs. like what I goggled:
`Uni.createFrom().item(1)
        .onItem().transform(i -> "hello-" + i)
        .onItem().delayIt().by(Duration.ofMillis(10000))`
This creates a Uni and delay some time. Then you may use a test endpoint to return it. Mutiny is very different coding style so you may spend some time to study it.

Reply:
In case of Gateway, the real action webclient sending the proxied request, e.g, "client.get( p, request ).call()"

Reply:
Thanks very much, let me have a try.